clients/src/main/java/org/apache/kafka/common/config/TopicConfig.java

```diff
@@ -74,7 +74,7 @@
         "their data.";
 
     public static final String MAX_MESSAGE_BYTES_CONFIG = "max.message.bytes";
-    public static final String MAX_MESSAGE_BYTES_DOC = "This is largest message size Kafka will allow to be " +
+    public static final String MAX_MESSAGE_BYTES_DOC = "This is largest record batch size Kafka will allow to be " +
```

Comment:
Hmm... Given the name of this config and the fact that we continue to support the v0/v1 formats, I wonder if we're ready to jump to the "record batch" language just yet. Maybe we could add something like this: "Messages are grouped in batches for efficiency. In previous releases, this applied only if compression was enabled, but it is generally true in the latest message format version. This configuration limits the maximum size of of a single uncompressed message for older formats and the size of a single message batch otherwise."